### PR TITLE
Topic/tombstone feature

### PR DESCRIPTION
**Pull request by HuwCampbell**
WIP.
@jystic Not sure where Jetski is supposed to be.
Also there's some java which I have put a todo on.

### CODE REVIEW

File: data/libs/prelude.icicle [hunk 1]
```diff
@@ -15,3 +15,6 @@ sd v = let fold a = ((0, 0.0), 0.0)
 # Time statements
 newest v = latest 1 ~> let fold1 s = v : v ~> s.
 oldest v = let fold1 s = v : s ~> s.
+
+# Boolean statements
+not a = let r = case a | True -> False | False -> True end ~> r.
```
Comment (tranma) [inline data/libs/prelude.icicle]: this can just be `not a = case a | True -> False | False -> True end.`

File: test/cli/repl/t20-lexer/expected [hunk 2]
```diff
@@ -30,7 +30,7 @@ Check error:
       mean                 : (Num a) => Element a -> Aggregate Double
       date                 : Element Definitely DateTime
       min                  : Element a -> Aggregate Possibly a
-      sd                   : (Element =: ReturnOfLet Pure a) => a Double -> Aggregate Double
+      not                  : (t$179 =: TemporalityJoin a Pure) => a Bool -> Pure Bool
```
Comment (tranma) [inline test/cli/repl/t20-lexer/expected]: doh, leftover constraints =(

File: data/libs/prelude.icicle (hunk 1 above)
Comment (HuwCampbell) [inline data/libs/prelude.icicle]: Yep, I actually already did that one.

**Comment by jystic** [comment]
@HuwCampbell it's fine if you upgrade it, the only difference is the merge commit, my bad for not doing this already.

**Comment by HuwCampbell** [comment]
@amosr would you mind having a look at this. I think someone else will have to do the Java generation part, but apart from that I think it is ok.

**Comment by amosr** [comment]
Yep, this looks good. I suspect this will all change in a little bit once I get the possibly shit going, but for now I'm happy to merge this.
Don't worry about the java generation either, that should fall out of the possibly stuff.

**Comment by amosr** [comment]
:children_crossing:

**Pull request merged**